analytics-api/app/controllers/JobController.scala

```diff
@@ -116,15 +123,68 @@ class JobController @Inject() (
       result("OK", JSONUtils.serialize(CommonUtil.OK(APIIds.CHANNEL_TELEMETRY_EXHAUST, Map("msg" -> s"$cacheType cache refreshed successfully"))))
   }
 
-  def authorizeDataExhaustRequest(consumerId: String, channelId: String): (Boolean, Option[String]) = {
+  def authorizeDataExhaustRequest(request: Request[AnyContent], authorizedRoles: List[String], superAdminRulesCheck: Boolean = false): (Boolean, Option[String]) = {
+
+    // security enhancements logic
+    val channelId = request.headers.get("X-Channel-ID").getOrElse("")
+    val consumerId = request.headers.get("X-Consumer-ID").getOrElse("")
+    val userAuthToken = request.headers.get("x-authenticated-user-token")
+    val authBearerToken = request.headers.get("Authorization")
+    val userApiUrl = config.getString("user.profile.url")
     if (channelId.nonEmpty) {
-        APILogger.log(s"Authorizing $consumerId and $channelId")
-        val whitelistedConsumers = config.getStringList("channel.data_exhaust.whitelisted.consumers")
-        // if consumerId is present in whitelisted consumers, skip auth check
-        if (consumerId.nonEmpty && whitelistedConsumers.contains(consumerId)) (true, None)
+        if(userAuthToken.isEmpty) {
+            APILogger.log(s"Authorizing $consumerId and $channelId")
+            val whitelistedConsumers = config.getStringList("channel.data_exhaust.whitelisted.consumers")
+            // if consumerId is present in whitelisted consumers, skip auth check
+            if (consumerId.nonEmpty && whitelistedConsumers.contains(consumerId)) (true, None)
+            else {
+                val status = Option(cacheUtil.getConsumerChannelTable().get(consumerId, channelId))
+                if (status.getOrElse(0) == 1) (true, None) else (false, Option(s"Given X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId' are not authorized"))
+            }
+        }
         else {
-            val status = Option(cacheUtil.getConsumerChannelTable().get(consumerId, channelId))
-            if (status.getOrElse(0) == 1) (true, None) else (false, Option(s"Given X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId' are not authorized"))
+            // get userId from user auth token
+            val userId = accessTokenValidator.getUserId(userAuthToken.get)
+            var unauthorizedErrMsg = "You are not authorized."
+            APILogger.log("userId retrieved: " + userId)
+            if(!"Unauthorized".equalsIgnoreCase(userId)) {
+                val headers = Map("x-authenticated-user-token" -> userAuthToken.get, "Authorization" -> authBearerToken.getOrElse(""))
```

Comment:
i don't think we need to send the bearer token if we are calling the service via internal end point

Reply:
For these changes give new PR

Reply:
Done - removed bearer token as we are using internal end point